### PR TITLE
Adrien/fixing ghpages

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
This PR allows to build the entire docs using the new Transifex CLI from within a container.
It also adds a workflow_dispatch for triggering manually for "emergency" cases.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
 
-CMD . .venv/bin/activate && mkdocs serve -a 0.0.0.0:8000
+#   Token pass through for docker build --build-arg ...
+ARG tx_token
+ENV TX_TOKEN $tx_token
+ENV DEFAULT_LANGUAGE_ONLY false
+
+#   Set up and fetch translations the docs
+RUN tx add --project qfield-documentation --file-filter 'documentation/<project_slug>.<resource_slug>/<lang>.<ext>' remote https://www.transifex.com/opengisch/qfield-documentation/dashboard/ && \
+    tx pull --all
```
Comment (suricactus) [inline Dockerfile:32]:
```suggestion
RUN tx add --project qfield-documentation --file-filter 'documentation/<project_slug>.<resource_slug>/<lang>.<ext>' remote https://www.transifex.com/opengisch/qfield-documentation/dashboard/ \
    && tx pull --all
```

I find it much more readable with operators on the new line, as you can clearly see it is a continuation from the previous line.

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,40 @@
-# Setup a Python virtual environment
+#   Setup a Python virtual environment
 FROM python:3.9.13-slim-bullseye as python-builder
```
Comment (suricactus) [inline Dockerfile:2]:
```suggestion
FROM python:3.9.13-slim-bullseye AS python-builder
```

File: Dockerfile [hunk 3]
```diff
@@ -1,24 +1,40 @@
...
+RUN tx add --project qfield-documentation --file-filter 'documentation/<project_slug>.<resource_slug>/<lang>.<ext>' remote https://www.transifex.com/opengisch/qfield-documentation/dashboard/ && \
+    tx pull --all
+
+#   Build the docs
+CMD . .venv/bin/activate && mkdocs build
+
+# To extract the build, run with a bindmount, for example:
+#   mkdir site && docker run -it -v ./site:/opt/app/site --rm localhost/qfield-docs
+# Then you can serve locally with your favorite web server, for example:
+#   python3 -m http.server -d ./site
```
Comment (suricactus) [inline Dockerfile:40]: Thins might need to be changed, probably using the native `mkdocs serve` from within the container is a better idea. Will see in a follow up PR, as we discussed.

Comment (why-not-try-calmer) [reply Dockerfile:40]: Absolutely!

**Review by suricactus — APPROVED** [review]

**Pull request merged**